book/src/volume-datasources.md

```diff
@@ -0,0 +1,15 @@
+## Kubernetes PVC DataSource (CSI VolumeContentSource)
+
+When creating a new PersistentVolumeClaim, the Kubernetes API provides a `PersistentVolumeClaim.DataSource` parameter.  This parameter is used to specify the CSI `CreateVolumeRequest.VolumeContentSource` option for CSI Provisioners The `VolumeContentSource` parameter instructs the CSI plugin to pre-populate the volume being provisioned with data from the specified source. 
+
+### External Provisioner Responsibilities
+
+If a `DataSource` is specified in the `CreateVolume` call to the CSI external provisioner, the external provisioner will evaluate whether the type of DataSource is supported, and if the data contained within the reference object is valid.  If the specified `DataSource.Kind` is supported, and the object is valid, the external provisioner will attempt to fetch the specified resource, ensuring that exists and obtaining the appropriate ID of the object to pass to the plugin.
```

Comment:
the last part can be simplified to "the external provisioner will fetch the resource and pass the appropriate object id to the plugin"